### PR TITLE
Don't Call Google Analytics Until It's Defined

### PR DESCRIPTION
**Pull request by tubbo**
Instead of seeing whether the app is in the test environment, check
whether the `ga()` function is present before sending any calls to it.
This prevents a JS error that occurs when the
`config.google_analytics_tracking_id` is missing.

### CODE REVIEW

**Comment by tubbo** [comment]
there's no workflow for this branch but i think adding one should be its own unit of work.

**Comment by tubbo** [comment]
@bencrouse builds shouldn't be setting the GA ID from config for a few reasons. for one thing, it makes tests run a lot slower since they now have to make an outbound HTTP call (which may or may not fail on the env, causing tests to break intermittently. this happened to me on a few past builds and it was extra irritating). but there's also no need to do it because it conflates data between staging and production, so if you're gonna set `config.google_analytics_tracking_id`, you should be doing it in **config/environments/production.rb**, not **config/application.rb**.

TLDR: it should _always_ be undefined in tests. if it's not, there's a problem with the app configuration.

**Comment by mttdffy** [comment]
I agree that in theory you shouldn't have google analytics configured for test, but in reality it will not be uncommon for it to happen without people realizing it. Like when they just set the config in the `workarea.rb` initializer.  We can't control how people set this value so we might as well check both conditions. There's no harm on our side.
